solr/core/src/java/org/apache/solr/search/grouping/distributed/responseprocessor/SkipSecondStepSearchGroupShardResponseProcessor.java

```diff
@@ -0,0 +1,116 @@
+  }
+
+  @Override
+  protected SearchGroupsContainer newSearchGroupsContainer(ResponseBuilder rb) {
+    return new SkipSecondStepSearchGroupsContainer(rb.getGroupingSpec().getFields());
+  }
+
+  @Override
+  public void process(ResponseBuilder rb, ShardRequest shardRequest) {
+    super.process(rb, shardRequest);
+    TopGroupsShardResponseProcessor.fillResultIds(rb);
+  }
+
+  protected static class SkipSecondStepSearchGroupsContainer extends SearchGroupsContainer {
+
+    private final Map<Object, String> docIdToShard = new HashMap<>();
+
+    public SkipSecondStepSearchGroupsContainer(String[] fields) {
+      super(fields);
+    }
+
+    @Override
+    public void addSearchGroups(ShardResponse srsp, String field, Collection<SearchGroup<BytesRef>> searchGroups) {
+      super.addSearchGroups(srsp, field, searchGroups);
+      for (SearchGroup<BytesRef> searchGroup : searchGroups) {
+        assert(srsp.getShard() != null);
+        docIdToShard.put(searchGroup.topDocSolrId, srsp.getShard());
+      }
+    }
+
+    @Override
+    public void addMergedSearchGroups(ResponseBuilder rb, String groupField, Collection<SearchGroup<BytesRef>> mergedTopGroups ) {
+      // TODO: add comment or javadoc re: why this method is overridden as a no-op
+    }
+
+    @Override
+    public void addSearchGroupToShards(ResponseBuilder rb, String groupField, Collection<SearchGroup<BytesRef>> mergedTopGroups) {
+      super.addSearchGroupToShards(rb, groupField, mergedTopGroups);
+
+      final GroupingSpecification groupingSpecification = rb.getGroupingSpec();
+      final Sort groupSort = groupingSpecification.getGroupSortSpec().getSort();
+
+      GroupDocs<BytesRef>[] groups = new GroupDocs[mergedTopGroups.size()];
+
+      // This is the max score found in any document on any group
+      float maxScore = 0;
+      int index = 0;
+
+      for (SearchGroup<BytesRef> group : mergedTopGroups) {
+        maxScore = Math.max(maxScore, group.topDocScore);
+        final String shard = docIdToShard.get(group.topDocSolrId);
+        assert(shard != null);
+        final ShardDoc sdoc = new ShardDoc();
+        sdoc.score = group.topDocScore;
+        sdoc.id = group.topDocSolrId;
+        sdoc.shard = shard;
+
+        groups[index++] = new GroupDocs<>(group.topDocScore,
+            group.topDocScore,
+            new TotalHits(1, TotalHits.Relation.EQUAL_TO), /* we don't know the actual number of hits in the group- we set it to 1 as we only keep track of the top doc */
+            new ShardDoc[] { sdoc }, /* only top doc */
+            group.groupValue,
+            group.sortValues);
+      }
+      TopGroups<BytesRef> topMergedGroups = new TopGroups<BytesRef>(groupSort.getSort(),
+          rb.getGroupingSpec().getWithinGroupSortSpec().getSort().getSort(),
+          0, /*Set totalHitCount to 0 as we can't computed it as is */
+          0, /*Set totalGroupedHitCount to 0 as we can't computed it as is*/
+          groups,
+          maxScore);
```

Comment:
https://github.com/cpoerschke/lucene-solr/commit/20129e7d3f7e12f442254e780e7da9a590a9036b proposes to initialise `maxScore` to `Float.MIN_VALUE` and to pass `Float.NaN` here if `groups` happened to be empty.

Reply:
THanks. merged

Reply:
ok

Reply:
Yes, I added a test to check that the result if ok in case of 0 results